syntax = "proto3";

message Todo{
    int32 id = 1;
    string content = 2;
    enum Status{
        COMPLETE = 0;
        PENDING = 1;
    }
    Status status = 3;
}

message AllTodos{
    repeated Todo todo = 1;
}

message Empty{

}

service TodoApp{
    rpc AddTodo (Todo) returns (Empty) {}
    rpc ListAllTodos (Empty) returns (AllTodos) {}
    rpc EditTodo (Todo) returns (Empty) {}
    rpc RemoveTodo (Todo) returns (Todo) {}
    rpc UpdateStatus (Todo) returns (Todo) {}
    rpc ListAllStream (Empty) returns (stream Todo) {}
    rpc GetTodo (Todo) returns (Todo) {}
}